gnucash/import-export/aqb/dialog-ab-select-imexporter.c

```diff
@@ -0,0 +1,245 @@
+    imexd = g_new0(GncABSelectImExDlg, 1);
+    imexd->parent = parent;
+    imexd->abi = abi;
+
+    g_signal_connect (parent, "destroy",
+                      G_CALLBACK (gtk_widget_destroyed), &imexd->parent);
+    builder = gtk_builder_new();
+    gnc_builder_add_from_file (builder, "dialog-ab.glade", "imexporter-list");
+    gnc_builder_add_from_file (builder, "dialog-ab.glade", "profile-list");
+    gnc_builder_add_from_file (builder, "dialog-ab.glade",
+                               "aqbanking-select-imexporter-dialog");
+    imexd->dialog =
+        GTK_WIDGET (gtk_builder_get_object (builder,
+                                            "aqbanking-select-imexporter-dialog"));
+    g_signal_connect (imexd->dialog, "destroy",
+                      G_CALLBACK (gtk_widget_destroyed), &imexd->dialog);
+    imexd->imexporter_list =
+        GTK_LIST_STORE (gtk_builder_get_object (builder, "imexporter-list"));
+    imexd->profile_list =
+        GTK_LIST_STORE (gtk_builder_get_object (builder, "profile-list"));
+    imexd->select_imexporter =
+        GTK_WIDGET (gtk_builder_get_object (builder, "imexporter-sel"));
+    imexd->select_profile =
+        GTK_WIDGET (gtk_builder_get_object (builder, "profile-sel"));
+    imexd->ok_button =
+        GTK_WIDGET (gtk_builder_get_object (builder, "imex-okbutton"));
+
+    imex_select = GTK_TREE_SELECTION (gtk_builder_get_object (builder, "imex-selection"));
+    prof_select = GTK_TREE_SELECTION (gtk_builder_get_object (builder, "prof-selection"));
+    populate_list_store (imexd->imexporter_list,
+                         imexporters);
+
+    g_signal_connect (imex_select, "changed", G_CALLBACK(imexporter_changed),
+                      imexd);
+    g_signal_connect (prof_select, "changed", G_CALLBACK(profile_changed),
+                      imexd);
+    g_list_free (imexporters);
+    g_object_unref (G_OBJECT (builder));
+
+    gtk_window_set_transient_for (GTK_WINDOW (imexd->dialog),
+                                  GTK_WINDOW (imexd->parent));
+
+    return imexd;
+}
+
+void
+gnc_ab_select_imex_dlg_destroy (GncABSelectImExDlg* imexd)
+{
+
+    if (imexd->imexporter_list)
+        gtk_list_store_clear (imexd->imexporter_list);
+
+    if (imexd->profile_list)
+        gtk_list_store_clear (imexd->profile_list);
+
+    if (imexd->dialog)
+        gtk_widget_destroy (imexd->dialog);
+
+    g_free (imexd);
+}
+
+gboolean
+imexporter_changed(GtkTreeSelection* sel, gpointer data)
+{
+    GncABSelectImExDlg* imexd = (GncABSelectImExDlg*)data;
+    GtkTreeIter iter;
+    GtkTreeModel* model;
+
+    gtk_widget_set_sensitive (imexd->ok_button, FALSE);
+
+    if (gtk_tree_selection_get_selected (sel, &model, &iter))
+    {
+        char* name = NULL;
+        GList* profiles = NULL;
+
+        gtk_tree_model_get (model, &iter, NAME_COL, &name, -1);
+        if (name && *name)
+        {
+             profiles = gnc_ab_imexporter_profile_list (imexd->abi, name);
+             g_free (name);
+        }
```

Comment:
To prevent leakage whereby `name==""`
```suggestion
        if (name && *name)
             profiles = gnc_ab_imexporter_profile_list (imexd->abi, name);
        g_free (name);
```

Reply:
Thanks!